src/api/api.ts

```diff
@@ -1 +1,31 @@
 export const BASE_URL = 'https://mate.academy/students-api';
+
+export const request = async (url: string) => {
+  const response = await fetch(`${BASE_URL}${url}`);
+
+  if (!response.ok) {
+    throw new Error(`${response.status} - ${response.statusText}`);
+  }
+
+  return response.json();
+};
+
+export const requestWithOptions = async (url: string, options: Option) => {
+  const response = await fetch(`${BASE_URL}${url}`, options);
+
+  if (!response.ok) {
+    throw new Error(`${response.status} - ${response.statusText}`);
+  }
+
+  return response.json();
+};
```

Comment:
these methods are almost identical, delete requestWithOptions function, instead just add an optional options argument to the request function

Reply:
Fixed